.travis.yml

```diff
@@ -30,7 +30,7 @@ notifications:
 
 env:
   global:
-   - MAVEN_OVERRIDE="--settings=.travis/settings.xml -Dmaven.javadoc.skip=true -Dcheckstyle.skip=true -Dfindbugs.skip=true"
+   - MAVEN_OVERRIDE="--settings=.travis/settings.xml"
```

Comment:
Fair enough, though with maven that ship has already sailed. One cannot, in general, quickly and concisely run a single target. Fighting the baked-in conceptual model of "run [a prefix of] everything in a particular order" has already backfired - IMO the number of profiles we have for managing different build modes, and the general lack of knowledge about which are important in which situations, indicates failure of the approach (which, to be transparent, I instigated, so I am recanting my prior position here). If someone has a general design pattern for simulating target/dependency-based builds in our mvn I would be very grateful but otherwise don't like where things have headed.

One note, since we used different syntax in our commentary here: I think `-D` should be favored over `-P` because you can activate a selection of profiles based on a property while you cannot activate a profile based on the activation of another profile. I believe it is strictly more powerful.

And less severely `-D precommit` as a name seems to tie the config too tightly with administrative policies, versus `-D full` or `-D thorough`. I actually like having one thing a user can specify that will do ~"the same" thing that happens on their PR", but as with my other preference above, the results of this approach so far has made me reconsider. The pitfall is that one might mistakenly think it is OK to put esoteric configuration in this profile. One mild example is how our `-P jenkins-precommit` sets up `--project` parameters that are not appropriate for a user.

All that said, LGTM :-)